### DISCUSSION TITLE
Location of ascii arts

### DISCUSSION THREAD

**Discussion by TanmayPatil105** [TanmayPatil105/procfetch#108] [Ideas]
Currently we are moving the ascii arts to `/usr/share/procfetch`.
Both manually and brew install does the same.
Can we do something like
- When installed manually, ascii arts will be moved to `/usr/share/procfetch`
- When installed using brew, ascii arts will be moved to `/home/linuxbrew/.linuxbrew/share`

We need to make changes to code according 

```diff
void print_process(string art, string color_name)
{
-    string path = "/usr/share/procfetch/ascii/" + art;
+    string path = "SOME_VARIABLE" + art;
    fstream fptr;
```

**Comment by akshaykhoje** [comment]
Yeah! Makes it safer not interfering with the root :)
Also, that's a better approach to keep a non-essential package separate from the essential ones. Need to look into this a bit more though.

**Comment by youpong** [comment]
Generate the appropriate build system for each of brew and manual install with arguments to the configure script.

For brew:
```
$ ./configure -C brew
```
For manually:
```
$ ./configure
```

**Reply by youpong** [reply]
`configure` script substitute @INSTALL_LIB_DIR@ with actual value.

config.h
```
#define INSTALL_LIB_DIR @INSTALL_LIB_DIR@
```

```
void print_process(string art, string color_name)
{
-    string path = "/usr/share/procfetch/ascii/" + art;
+    string path = INSTALL_LIB_DIR + "/" + art;
    fstream fptr;
```

**Reply by youpong** [reply]
ascii/Makefile.in:
```
DATADIR = @INSTALL_LIB_DIR@
```

**Reply by TanmayPatil105** [reply]
Yupp, this is what I meant.
And need to make changes [here](https://github.com/TanmayPatil105/homebrew-tap/blob/main/procfetch.rb) to:
```
class Procfetch < Formula
    desc "🎨 A command-line system information utility written in C++"
    homepage "https://tanmaypatil105.github.io/procfetch/"
    url "https://github.com/TanmayPatil105/procfetch/archive/refs/tags/v0.1.2.tar.gz"
    sha256 "7788bcbb666337499ed00088b66948e74aed6cb743030d82eb2dd1a65569d710"
    license "GPL-3.0"
    
    def install
      system "./configure -c brew"
      system "make"
      system "sudo mkdir -p $HOMEBREW_PREFIX/share/procfetch/ascii"
      system "sudo cp ascii/* $HOMEBREW_PREFIX/share/procfetch/ascii/"
      bin.install "src/procfetch"
    end
  
    test do
      system "procfetch -t"
    end
  end
```

**Reply by TanmayPatil105** [reply]
Changing `configure` to something like
```
if [[ $# != 0 && $1 == "-c" && $2 == "brew" ]];
then
    echo "#define INSTALL_LIB_DIR \"$HOMEBREW_PREFIX/procfetch/ascii\"" >> src/config.h
else
    echo "#define INSTALL_LIB_DIR \"/usr/share/procfetch/ascii\"" >> src/config.h
fi 
```
Ik here I am checking so many conditions :p

**Comment by youpong** [comment]
Done on `configure` script and C++ sources.  
* #113
* #118

Homebrew work remains to be done.

**Comment by youpong** [comment]
The Homebrew work have done.

* TanmayPatil105/homebrew-tap#1 

So, All work have finished.
